pkg/controller/constraint/constraint_controller_test.go

```diff
@@ -91,3 +95,250 @@ func TestHasVAPCel(t *testing.T) {
+			enforcementAction:             util.Dryrun,
+			defGenerateVAPB:               true,
+			instance:                      &unstructured.Unstructured{},
+			expectedGenerate:              true,
+			expectedVAPEnforcementActions: []string{"dryrun"},
+		},
+		{
+			name:              "defaultGenerateVAPB is false, enforcementAction is Scoped, VAP ep is not set",
+			enforcementAction: util.Scoped,
+			defGenerateVAPB:   false,
+			instance: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"spec": map[string]interface{}{
+						"enforcementAction": "scoped",
+						"scopedEnforcementActions": []apiconstraints.ScopedEnforcementAction{
+							{
+								Action: "deny",
+								EnforcementPoints: []apiconstraints.EnforcementPoint{
+									{
+										Name: util.WebhookEnforcementPoint,
+									},
+								},
+							},
+							{
+								Action: "warn",
+								EnforcementPoints: []apiconstraints.EnforcementPoint{
+									{
+										Name: util.WebhookEnforcementPoint,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedGenerate:              false,
+			expectedVAPEnforcementActions: []string{},
+		},
+		{
+			name:              "defaultGenerateVAPB is true, enforcementAction is Scoped, VAP ep is not set",
+			enforcementAction: util.Scoped,
+			defGenerateVAPB:   true,
+			instance: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"spec": map[string]interface{}{
+						"enforcementAction": "scoped",
+						"scopedEnforcementActions": []apiconstraints.ScopedEnforcementAction{
+							{
+								Action: "deny",
+								EnforcementPoints: []apiconstraints.EnforcementPoint{
+									{
+										Name: util.AuditEnforcementPoint,
+									},
+								},
+							},
+							{
+								Action: "warn",
+								EnforcementPoints: []apiconstraints.EnforcementPoint{
+									{
+										Name: util.AuditEnforcementPoint,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedGenerate:              false,
+			expectedVAPEnforcementActions: []string{},
+		},
+		{
+			name:              "defaultGenerateVAPB is false, enforcementAction is Scoped, VAP ep is set",
+			enforcementAction: util.Scoped,
+			defGenerateVAPB:   false,
+			instance: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"spec": map[string]interface{}{
+						"enforcementAction": "scoped",
+						"scopedEnforcementActions": []apiconstraints.ScopedEnforcementAction{
+							{
+								Action: "deny",
+								EnforcementPoints: []apiconstraints.EnforcementPoint{
+									{
+										Name: util.WebhookEnforcementPoint,
+									},
+									{
+										Name: util.VAPEnforcementPoint,
+									},
+								},
+							},
+							{
+								Action: "warn",
+								EnforcementPoints: []apiconstraints.EnforcementPoint{
+									{
+										Name: util.WebhookEnforcementPoint,
```

Comment:
in this test, when webhookep has both action warn and deny, which one wins?

Reply:
if both the actions are there ultimatly the request is denied but output containes warning and deny and looks like this - 

```
Warning: [cm-must-have-gk-scoped] you must provide labels: {"gatekeeper"}
Error from server (Forbidden): error when creating "/mount/d/go/src/github.com/open-policy-agent/gatekeeper/test/bats/tests/bad/bad_cm_scoped.yaml": admission webhook "validation.gatekeeper.sh" denied the request: [cm-must-have-gk-scoped] you must provide labels: {"gatekeeper"}
```